test/models/watching_transaction_test.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class WatchingTransactionTest < ActiveSupport::TestCase
+  test 'create new watching transaction should work' do
+    params = attributes_for(:watching_transaction)
+
+    tx = WatchingTransaction.new(params)
+
+    user = create(:user)
+    tx.user_id = user.id
+
+    assert tx.save!,
+           'should save'
+
+    assert_not WatchingTransaction.new(params).save,
+               'should fail with the same data'
+  end
+
+  test 'watch should work' do
+    user = create(:user)
+    attrs = attributes_for(:watching_transaction)
+    ok, tx = WatchingTransaction.watch(user, attrs)
+
+    assert_equal :ok, ok,
+                 'should work'
+  end
+
+  test 'resend should work' do
+    user = create(:user)
+    watching_transaction = create(:watching_transaction, user: user)
+    attrs = attributes_for(:watching_transaction)
+
+    ok, tx = WatchingTransaction.resend(user, watching_transaction, attrs)
+
+    assert_equal :ok, ok,
+                 'should work'
+  end
+
+  test 'resend_transactions' do
```

Comment:
No need to compact all conditions on one test. You can write separate tests for each property to better glance it:

```
# For each test, create random groups sizes as much as possible.
# Perhaps, create a randomly sized group of transactions of at least 3 to at most 7

test "resend transaction should work" do
  # This tests a transaction group if every transaction is dropped
  # Then only the latest one is sent 
  # Assert `get_transaction` was called n times and `send_transaction` 1 time.
  # Assert no record was deleted
end

test "resend transaction should work if any is mined"" do
  # This tests if any transaction is mined,
  # Then the whole transaction group is deleted
end

test "resend transaction should delete dropped transactions" do
  # This tests if any transaction is dropped,
  # Then those transaction is deleted/destroyed
  # In this test, the web mock should have a dropped response to at least one transaction and those transactions are deleted.
  # However, at least one transaction should be pending so that no resent transaction is sent.
end 

test "resend transaction should work when API is down" do
  # Assuming with everything else works,
  # Test if `get_transaction` returns a HTTP error and test if `send_transaction` also returns an error.
  # It should be safe and error free.
end
```